### ISSUE TITLE
Plot/figure brainstorming for cross-site comparisons

### ISSUE THREAD

**Issue by evaalonsoortiz** [spinal-cord-7t/coil-qc-code#7]
Our dataset incudes 7 sites with 3 subjects per-site (the same 3 subjects were, or are to be, scanned across all 7 sites). The goal is to compare coil designs. The purpose of this issue is to develop a consensus on what metrics we want to derive from the multi-subject data and how we want to visualize those metrics across sites.

Please comment on whether or not you agree with the "goals" and the proposed figures. Alternatively, please suggest and "goal" and/or figure for a given scan type.

Images:

------------------

**TFL_B1_C3C4** --->  B1+ map acquired with the Optimal Reference Voltage

_Goals:_ 

- Assess B1+ efficiency along the SC
- Assess B1+ uniformity along the SC

_Suggested figures:_
Figure 1:
<img width="600" alt="Screenshot 2023-12-04 at 9 30 09 PM" src="https://github.com/spinal-cord-7t/coil-qc-code/assets/6652194/21bb5c76-f818-4e7d-a311-97d1f5b97c6a">

Figure 1 shows us how uniform B1+ is along the SC but it does not tell us how efficient the coil is. For that we may want to include the following figure:

Figure 2:
<img width="600" alt="Screen Shot 2023-12-18 at 3 54 53 PM" src="https://github.com/spinal-cord-7t/coil-qc-code/assets/6652194/8d047569-2589-4ae2-b08e-bc4641712d62">

------------------

**COILQA_SAG_LARGE** (H>F phase encoding)

Images: SNR, g-factor, correlation matrix

_Goals:_

- Assess SNR along the SC
- Assess SNR uniformity along the SC 
- g-factor values in the SC
- Correlation matrix assessment 

_Suggested figures:_

For SNR: same as those for TFL (B1+)

For g-factor: 

Using an enlarged spinal cord (SC) ROI, compute the average g-factor along the SC report it along with the standard deviation. 

Figure 3:
<img width="600" alt="Screenshot 2023-12-18 at 4 58 36 PM" src="https://github.com/spinal-cord-7t/coil-qc-code/assets/6652194/38b20b15-6bfb-40d6-a710-8a7f97924da1">

The drawback of Figure 3 is that the little bars for each subject might overlap quite a bit making it overall hard to read. Alternatively we can represent the g-factor along the SC for each subject as a histogram/distribution in the form of a ridge-plot:

Figure 4:
<img width="600" alt="Screenshot 2023-12-18 at 5 00 40 PM" src="https://github.com/spinal-cord-7t/coil-qc-code/assets/6652194/9cf50ce4-ee99-4118-a919-da5085896927">

Figures 3 or 4 can be a sub-plot for a larger figure that includes other acceleration factors (3x3, 3x2).

For correlation matrices: 

Compute the average correlation matrix across subjects for each site and show the average matrix for each of the 7 sites in a tiled arrangement. 

------------------

**COILQA_SAG_SMALL** (H>F phase encoding)

_Goals:_

- g-factor values in the SC

_Suggested figures:_

Same as Figure 3 or 4. This could be included in supporting documents.

------------------

**COILQA_TRA** (L>R phase encoding)

_Goals:_

- Assess SNR uniformity within the SC (in-plane)
- g-factor values in the SC

_Suggested figures:_

For SNR: Compute in-plane standard-deviation and average that for all slices. Show this value for each subject, and across sites, like in Figure 1.

For g-factor: Same as Figure 3 or 4. 

------------------

**DREAM_LARGE**

_Goals:_

- DREAM has little DC bias in areas where the refV is well-calibrated, but a narrow range of refV where it is well behaved. The large FOV scan (acquired at the same refV as TFL) can be used as a second measurement, to corroborate the measured TFL values.

_Suggested figures:_

Same as Figure 2. Note: A bias in measured B1+ where the refV is not well calibrated can also lead to differences in measured B1+ uniformity. Therefore we may want to have another version of Figure 1. 

------------------

**DREAM_MEDIUM** + **DREAM_MEDIUM_066** + **DREAM_MEDIUM_HWLIMIT**

_Goals:_

- DREAM has little DC bias in areas where the refV is well-calibrated, but a narrow range of refV where it is well behaved. Compare measured B1+ efficiency (analogous to FA) across different refV values. 

_Suggested figures:_

Compute the average B1+ along the spinal cord and average across subjects, for each refV. The former will be included in a tiled figure where each sub-plot represents one site. 

Figure 5:
<img width="535" alt="Screenshot 2023-12-18 at 6 33 04 PM" src="https://github.com/spinal-cord-7t/coil-qc-code/assets/6652194/0fb5a9f5-0726-4486-a8ba-f2e9d229eb43"> 

Each site has 3 violin plots (refV, 0.66refV, 1.5refV) containing the distribution of B1+ values for all subjects.

Figure 6:
<img width="581" alt="Screenshot 2023-12-18 at 6 34 31 PM" src="https://github.com/spinal-cord-7t/coil-qc-code/assets/6652194/b067e1c3-851b-4ef8-a3f6-fbd125d33cef">

------------------

**GRE** ---> uncombined GRE images

_Goals:_

- characterize individual channels

_Suggested figures:_

A tiled figure (containing each channel's GRE magnitude) could be generated for each subject at each site and presented in supporting documentation.

**Comment by jcohenadad** [comment]
Thank you for the detailed figure layouts-- it makes it easier to comment. 

- TFL_B1_C3C4:
  - B1+: I would not average per slice and then compute the SD across slices, because the mean is sensitive to outliers. Instead, I would simply compute the SD across all voxels in the ROI (ie: SC mask), and 'play around' with ways to mitigate the presence of outliers (eg: remove 'noisy' voxels from the ROI before computing the SD-- we need to find a good way to remove those voxels-- playing the the code and the data, I will get better ideas how to do it).
  - I find it more informative to show the evolution of B1+ along the cord vs. the efficiency. So for B1+ I would have a panel A showing the actual maps for a representative subject, panel B showing the slice-wise mean across the SC for that subject, and panel C showing the SD for each subject. Alternatively, for panel C, we can consider showing the violin plot of B1+, and report the SD in a table instead.
  - efficiency: I would just represent the mean efficiency inside the ROI, for each subject and each site (again, making sure 'noisy' voxels are not included)
- COILQA_SAG_LARGE
  - SNR: People really like to see SNR maps, so I would also show one for a representative subject, and then show the mean SNR in the ROI along the cord for one subject, and the mean for all subjects (eg: the same panels A B and C as suggested above for B1+). That way people will also be able to assess uniformity.
  - g-factor: 
    - People commonly look at g-factor maps across various acceleration factors, because it gives a better sense of where g-factor is increased in the image, how the pattern looks like, etc. I would not reduce the information with numbers in an ROI. So I would just pick a representative subject and show maps of R=2x2, 2x3, 3x2, 3x3
    - Another consideration about the gfactor: depending how "large" the FOV is, I would not show the g-factor if there is too much background along the phase encoding direction. So I would just pick the most appropriate g-factor maps and show them in the paper. Happy to help pick the most appropriate one (we can open another issue and show all the acquisitions, and we can brainstorm there)
  - correlation matrix: I would not show these-- they are too 'low level' and are usually more a quality assessment of a given coil, rather than a performance assessment (which is what this paper is about). 
- DREAM_LARGE: same as the B1+ fig with panel A/B/C
- DREAM_MEDIUM + DREAM_MEDIUM_066 + DREAM_MEDIUM_HWLIMIT:
  - I would go with fig 5 (no violin plot-- it becomes too overwhelming)
- GRE
  - I think that the tiled figure with each individual channel will be overwhelming (to make and to look at), and also this is quite a 'low level' assessment. The bigger picture is assessed with the distribution of SNR along the cord
  - Instead, I would show representative images of T2* across sites (eg: one site per column, one level per row)

**Comment by evaalonsoortiz** [comment]
- TFL_B1_C3C4:
   * B1+: noted. We also have open the option of computing the median within the slice and then the SD (or the IQR) across slices. 
   *  :thumbsup:
   * Efficiency: noted. This could be in a table.
- COILQA_SAG_LARGE:
   * SNR: noted
   * g-factor: 
        * This would be a lot of visual data though (7 sites times  R=2x2, 2x3, 3x2, 3x3). How about picking a one acceleration only and showing it across sites for one subject? The rest can go in supplementary info. Unless @jvannesjo can think of another way to condense information visually?
       * For this image, the acceleration is along the H>F direction, so there is little background to corrupt the measurement. Otherwise, yes, I agree on picking just one.
   * correlation matrix: noted
- DREAM_LARGE: :thumbsup:
- DREAM_MEDIUM + DREAM_MEDIUM_066 + DREAM_MEDIUM_HWLIMIT: :thumbsup:
- GRE
   * Shall we not show this data at all then?
   * We did not acquire mGRE for this project. I could look at what was acquired for the qMRI project and we could discuss the with the other members of that project. We could further discuss that in a group meeting.

**Comment by jcohenadad** [comment]
> This would be a lot of visual data though (7 sites times R=2x2, 2x3, 3x2, 3x3). How about picking a one acceleration only and showing it across sites for one subject?

Yes, this is also what I had in mind. Showing 2x2 seems reasonable.

> GRE / Shall we not show this data at all then?

I would absolutely show at least one anat image. This is what most (ie not coil experts) people will want to see. If we don't have mGRE, GRE will suffice.

**Comment by evaalonsoortiz** [comment]
> > GRE / Shall we not show this data at all then?
> 
> I would absolutely show at least one anat image. This is what most (ie not coil experts) people will want to see. If we don't have mGRE, GRE will suffice.

It would still be a very big figure: all-coil images for one subject times 7 sites. I think supplementary figs is a good place for that, do you agree/disagree?

**Comment by jcohenadad** [comment]
if we pick one subj, one slice, it’s only 7 images, unless i am missing something?

**Comment by evaalonsoortiz** [comment]
We have 1 image per coil element (not coil combined). We can coil combine but we had agreed at one point to keep them uncombined in order to assess the coil elements themselves.

**Comment by evaalonsoortiz** [comment]
Summary of the agreed-upon figures:

**sub-xxx_acq-famp_TB1TFL**

Manuscript figure: 3 panel figure with (A) single subject representative B1+ map, (B) median B1+ within spinal cord ROI vs. spinal cord level for that subject, (C) median within slice (and ROI) and then inter-quartile range (IQR) across slices

Notebook figures:

- Figure 1: median within slice (and ROI) and then inter-quartile range (IQR) across slices 
- Figure 2: median B1+ within spinal cord ROI vs. spinal cord level for a single subject (the selected subject can be changed within the code)

_Note1: Note that for Figure 1, @jcohenadad wanted to explore how the data looks here, so this figure may evolve. We can play around with it once we have the code._
_Note2: TB1TFL maps are output as a flip angle map in units of flip angle * 10 (in degrees), to convert to nT/V use the same code as in https://github.com/shimming-toolbox/rf-shimming-7t/blob/main/data_processing.ipynb_

**sub-xxx_acq-refv_TB1DREAM**

Manuscript figure: same as for famp_TB1TFL

Notebook figures:

- Figure 3: same as Figure 1
- Figure 4: same as Figure 2

_Note1: DREAM outputs flip angle maps in degrees and B1+ maps that are normalized and in percent units_
_Note2: the 3 reference voltage acquisitions have to be combined into one in the following way:_

- _For each reference voltage a mask must be generated by thresholding the flip angle (FA) map to 20deg < FA < 50deg. The masks should then be applied to the corresponding B1+ maps. The three B1+ maps must be combined by averaging non-zero estimates of B1+ in each pixel._

_Note3: The FOV of the DREAM scans are smaller than the FOV of the TFL scans, therefore a one-to-one comparison between Figures 1 and 3, or Figures 2 and 4 can not be easily made_

**sub-xxx_acq-coilQaSagLarge_SNR**

Manuscript figure description: 3 panel figure with (A) single subject representative SNR map, (B) median SNR within spinal cord ROI vs. spinal cord level for that subject, (C) median SNR within spinal cord ROI vs. spinal cord level averaged across subjects per site
@jcohenadad I would argue here that B is not necessary and just keep A and C, let me know what you think.

Notebook figures:

- Figure 5: median B1+ within spinal cord ROI vs. spinal cord level for a single subject (the selected subject can be changed within the code)
- Figure 6: median B1+ within spinal cord ROI vs. spinal cord level averaged across subjects for each site

**Comment by jcohenadad** [comment]
some comments:
- we should make a figure showing mean (or median) B1+ efficiency along the cord, as done in the RF shimming paper, as a sanity check (a.k.a. 'look at your data before aggregating into statistics'). In the RF shimming paper, there was one column of plots, and each row was a subject. We could do the same, but add one column per site.
- fig 3/4 are the same as fig 1/2, hence the code should not be duplicated. Instead, the cell that generates the figure should loop across various B1+ methods. Create a list of B1+ methods and loop across that list.
- I would not address everything in one PR. I would start with B1+ data and then move on to the other items.

**Comment by jcohenadad** [comment]
> @jcohenadad I would argue here that B is not necessary and just keep A and C, let me know what you think.

Agreed. Single-subject statistics could go over the map, as done in the RF shimming paper, eg:

![image](https://github.com/spinal-cord-7t/coil-qc-code/assets/2482071/2845fe31-d4e3-4b78-be99-29d06afa866c)
